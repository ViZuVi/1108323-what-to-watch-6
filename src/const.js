export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  MOVIES: `/films`,
  LOGIN: `/login`,
  MOVIE_PAGE: `/films/:id`,
  MY_LIST: `/mylist`,
  LOGOUT: `/logout`
};


