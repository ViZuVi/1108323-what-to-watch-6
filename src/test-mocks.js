export const movies = [{
  id: 1,
  title: `Macbeth`,
  backgroundColor: `#F1E9CE`,
  backgroundImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  director: `Justin Kurzel`,
  genre: `Drama`,
  isavorite: false,
  posterImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 3.3,
  released: 2015,
  runTime: 113,
  scores: 48798,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
},
{
  id: 2,
  title: `Macbeth`,
  backgroundColor: `#F1E9CE`,
  backgroundImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  director: `Justin Kurzel`,
  genre: `Drama`,
  isavorite: false,
  posterImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 3.3,
  released: 2015,
  runTime: 113,
  scores: 48798,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
}];

export const movie = {
  id: 1,
  title: `Macbeth`,
  backgroundColor: `#F1E9CE`,
  backgroundImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  director: `Justin Kurzel`,
  genre: `Drama`,
  isavorite: false,
  posterImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 3.3,
  released: 2015,
  runTime: 113,
  scores: 48798,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
};

export const comments = [
  {
    id: 1,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
];

export const genres = [`All genres`, `Drama`, `Comedy`];

export const userInfo = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

