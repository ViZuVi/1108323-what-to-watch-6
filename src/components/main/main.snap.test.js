import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import Main from './main';
import {movies, genres, movie} from '../../test-mocks';
import {AuthorizationStatus} from '../../const';

const mockStore = configureStore({});

it(`Main should render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    DATA: {filteredMovies: movies, promoMovie: movie, isVisibleShowMore: true, genres},
    USER: {authorizationStatus: AuthorizationStatus.AUTH, userInfo: {}}
  });
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <Main onAddToFavoriteBtnClick={jest.fn()} />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
