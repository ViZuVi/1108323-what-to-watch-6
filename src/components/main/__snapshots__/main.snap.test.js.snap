// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
<div>
  <section
    class="movie-card"
  >
    <div
      class="movie-card__bg"
    >
      <img />
    </div>
    <h1
      class="visually-hidden"
    >
      WTW
    </h1>
    <header
      class="page-header movie-card__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="user-block"
      >
         
        <a
          href="/mylist"
        >
          <div
            class="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              width="63"
            />
          </div>
        </a>
        <a
          href="#"
        >
          Log out
        </a>
      </div>
    </header>
    <div
      class="movie-card__wrap"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster"
        >
          <img
            height="327"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          />
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            />
            <span
              class="movie-card__year"
            />
          </p>
          <div
            class="movie-card__buttons"
          >
            <a
              class="btn btn--play movie-card__button"
              href="/player/1"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog"
    >
      <h2
        class="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        class="catalog__genres-list"
      >
        <li
          class="catalog__genres-item"
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          class="catalog__genres-item"
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Drama
          </a>
        </li>
        <li
          class="catalog__genres-item"
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Comedy
          </a>
        </li>
      </ul>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/1"
          >
            <div
              class="small-movie-card__image"
            >
              <img
                alt="Shutter Island"
                height="175"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/1"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/2"
          >
            <div
              class="small-movie-card__image"
            >
              <img
                alt="Shutter Island"
                height="175"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg"
                width="280"
              />
            </div>
          </a>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/2"
            >
              Macbeth
            </a>
          </h3>
        </article>
      </div>
      <div
        class="catalog__more"
      >
        <button
          class="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
